pkg/service/node.go

```diff
@@ -236,34 +251,67 @@ func (n *NodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	if err != nil && !os.IsNotExist(err) {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
-	if !notMnt {
-		klog.V(3).Infof("Volume %s already mounted on %s", req.GetVolumeId(), targetPath)
-		return &csi.NodePublishVolumeResponse{}, nil
+	if notMnt {
+		if err := n.mountToTargetPath(req, block, targetPath, device, fsType, mountOptions); err != nil {
+			return nil, err
+		}
+	}
+
+	if !block {
+		if err := n.resizeFs(device, targetPath); err != nil {
+			return nil, err
+		}
 	}
 
-	if block {
-		err = n.ensureMountFileExists(targetPath)
+	return &csi.NodePublishVolumeResponse{}, nil
+}
+
+func (n *NodeService) resizeFs(device device, targetPath string) error {
+	ok, err := n.resizer.NeedResize(device.Path, targetPath)
+	if err != nil {
+		return status.Errorf(codes.Internal,
+			"need resize check failed on devicePath %s and targetPath %s, error: %v",
+			device.Path,
+			targetPath,
+			err)
+	}
+	if !ok {
+		// no resize is required
+		return nil
+	}
+	ok, err = n.resizer.Resize(device.Path, targetPath)
+	if !ok || err != nil {
+		return status.Errorf(codes.Internal,
+			"resize failed on path %s, error: %v", targetPath, err)
+	}
+
+	return nil
+}
+
+func (n *NodeService) mountToTargetPath(req *csi.NodePublishVolumeRequest, isBlock bool, targetPath string, device device, fsType string, mountOptions []string) error {
+	if isBlock {
+		err := n.ensureMountFileExists(targetPath)
 		if err != nil {
-			return nil, err
+			return err
 		}
 	} else {
 		// MkdirAll returns nil if path already exists
-		err = n.dirMaker.Make(targetPath, 0750)
+		err := n.dirMaker.Make(targetPath, 0750)
 		if err != nil {
-			return nil, err
+			return err
 		}
 		klog.V(3).Infof("GetMount() %v", req.VolumeCapability.GetMount())
 		klog.V(3).Infof("Mounting devicePath %s, on targetPath: %s with FS type: %s",
 			device, targetPath, fsType)
 	}
 
-	err = n.mounter.Mount(device.Path, targetPath, fsType, mountOptions)
+	err := n.mounter.Mount(device.Path, targetPath, fsType, mountOptions)
```

Comment:
Same just scope the err variable to the if statement.

Reply:
yup